bluesky/protocols.py

```diff
@@ -31,12 +53,39 @@ def success(self) -> bool:
 
 
 @runtime_checkable
-class Readable(Protocol):
+class Named(Protocol):
+    @property
+    def name(self) -> str:
+        ...
+
     @property
     def parent(self) -> Optional[Any]:
         """``None``, or a reference to a parent device."""
         ...
 
+
+@runtime_checkable
+class Configurable(Protocol):
+    def read_configuration(self) -> SyncOrAsync[Dict[str, EventData]]:
+        """Same API as ``read`` but for slow-changing fields related to configuration.
+
+        (e.g., exposure time)
+
+        These will typically be read only once per run.
+
+        Of course, for simple cases, you can effectively omit this complexity
+        by returning an empty dictionary.
+        """
+        ...
+
+    def describe_configuration(self) -> SyncOrAsync[Dict[str, EventDescriptor]]:
+        """Same API as ``describe``, but corresponding to the keys in
+        ``read_configuration``."""
+        ...
+
+
+@runtime_checkable
+class Readable(Named, Protocol):
```

Comment:
This separates Configurable and Readable. This is because simple objects (like Signals) don't have any configuration, so this saves a bit of code. I also plan to put `configure()` and `validate()` in `Configurable` at a later date

Reply:
I think I remember a question arising in Queueserver on whether we wanted bare Signals to show up as proper Readables or not, and the answer at that time being "no"

Opinions may have changed since then/queueserver can choose to further filter I suppose.

Reply:
Ophyd Signals at the moment conforms to Readable, although I don't know if the current behaviour of returning identical dictionaries for `read()` and `read_configuration()` is desirable:
https://github.com/bluesky/ophyd/blob/master/ophyd/signal.py#L381-L444

Maybe @dmgav could answer how Queueserver treats current Ophyd Signals?

Reply:
Decided that bare Signals don't need a read() method, but we should be able to get a Readable ability on any Signal via its top level Ability. E.g. "pilatus_NDStats_intensity" is produced from the pilatus Ability with a couple of special method calls to drill down

Reply:
decided split was fine